src/adapter/src/catalog/migrate.rs

```diff
@@ -60,9 +61,12 @@ pub(crate) async fn migrate(catalog: &mut Catalog) -> Result<(), anyhow::Error>
     // it. You probably should be adding a basic AST migration above, unless
     // you are really certain you want one of these crazy migrations.
     let cat = Catalog::load_catalog_items(&mut tx, catalog)?;
-    let _conn_cat = cat.for_system_session();
+    let conn_cat = cat.for_system_session();
 
-    rewrite_items(&mut tx, |_item| Ok(()))?;
+    rewrite_items(&mut tx, |item| {
+        normalize_create_secrets(&conn_cat, item)?;
```

Comment:
Unfortunately, I think that if a broken secret was committed to the stash, then it will be broken forever. For example, consider the secret: 
  `materialize.public.1.00`. 
Is the schema name `public` and the secret name `1.00` or is the schema name `public.1` and the secret name `00`? There's no way for us to correctly parse and normalize this.

Reply:
Indeed , the Nightly upgrade tests are now failing with:

```
environmentd: cannot migrate from catalog version 0.43.0 to version 0.44.0-dev (earlier versions might still work): Expected identifier, found operator "-"
```

which I assume is a demonstration of the impossibility to have a migration that fixes things?

Reply:
@philip-stoev  Can we detect the base version in the check itself? If so, we could not create such secrets for older base versions.

Reply:
No, this is currently not possible out of the box. [This ](https://github.com/MaterializeInc/materialize/issues/17636) is the ticket about creating the required enhancement to the testing framework.

Reply:
Oh right, I even assigned it to myself without reading it properly yet. Will work on that after the buildkite annotations.

Reply:
No need to disable anything I think, since as soon as the new version is released this week the test will start passing again.

Reply:
The magic of cloud. I'm still getting used to not having to support 3 year old versions. ;)